sympy/core/mul.py

```diff
@@ -853,9 +852,10 @@ def as_real_imag(self, deep=True, **hints):
             elif r.is_zero:
                 coeffi.append(i*S.ImaginaryUnit)
             elif a.is_commutative:
+                aconj = a.conjugate()
```

Comment:
Turned out that this is not possible when `other` is empty...

Reply:
That in turn is caused by this:
```python
In [6]: CRootOf(x**4 + x - 1, 2).expand(complex=True)
---------------------------------------------------------------------------
PolynomialError
```

Reply:
I'm not even sure how to disable that given the way that `Expr._expand_hint` is written:
https://github.com/sympy/sympy/blob/dff24a1f80d0704f18fd71a4875b5134a1c0edbf/sympy/core/expr.py#L3577-L3604
Here we have a CRootOf which is Expr and whose args are Expr. However it doesn't make any sense with a CRootOf to apply expansion to the args because they are already as expanded as they could be. The intent of `expand_complex` is to split into real and imaginary parts but with a CRootOf you don't do that by splitting the args into real and imaginary parts.

Clearly then what we need is for CRootOf to prevent its args from being expanded. However `Expr._expand_hint` recurses by always calling `Expr._expand_hint` meaning that no subclass can override it. It also always expands the args before calling any method of self. I don't see what method `CRootOf` could possibly define to be able to override this behaviour.